WordPress/gutenberg
# Fluid Typography: Fix bug in global styles where fluid clamp rules were not calculated for custom values

Fixes https://github.com/WordPress/gutenberg/issues/44752



## What?


Fix an issue as raised in https://github.com/WordPress/gutenberg/issues/44758 where the fluid typography clamp-based rules are not calculated for custom font-size values set at the block level in theme.json or in global styles.

## Why?


It was expected behaviour for the feature that adjusting custom font-size values at the block level in theme.json and global styles would result in the `clamp()` based rules being applied.

## How?


* In the Theme JSON PHP class, before declaring the `font-size` rule, pass the value through `gutenberg_get_typography_font_size_value`.
* For rendering realtime updates in the site editor, add a corresponding call to `getTypographyFontSizeValue` before returning the output in `getStylesDeclarations`

## Testing Instructions





Sample `theme.json` file:

```json
{
	"version": 2,
	"settings": {
		"appearanceTools": true,
		"layout": {
			"contentSize": "840px",
			"wideSize": "1100px"
		},
		"typography": {
			"fluid": true
		}
	}
}
```

1. With a blocks-based theme (such as emptytheme), enable fluid typography by updating `settings.typography.fluid` to `true` in the `theme.json` file.
2. In global styles, go to set the font-size for a particular block, e.g. Site Title, and set it to a custom font size.
3. Observe in the site editor that if you resize the window, the font-size should be fluid.
4. Save, and load the site front end. Check that the same rules are applied there.
5. Open up the post editor and insert a block that you styled in global styles. Its styling should be the same in the post editor.
6. If you switch off `settings.typography.fluid` in `theme.json`, the the fluid rules should no longer applied in the site editor or on the site frontend.

## Screenshots or screencast 

![2022-10-07 12 22 55](https://user-images.githubusercontent.com/14988353/194446751-afd12f96-004b-4adb-85b4-9eb77600fbc4.gif)
